Makefile

```diff
@@ -1,181 +1,55 @@
-#  dependencies
-
-# If go get is run from inside the project directory it will add the dependencies
-# to the go.mod file. To avoid that we change to a directory without a go.mod file
-# when downloading the following dependencies
-
-$(GOX):
-	(cd /; GO111MODULE=on go get -u github.com/mitchellh/gox)
-
-$(GOIMPORTS):
-	(cd /; GO111MODULE=on go get -u golang.org/x/tools/cmd/goimports)
-
-# ------------------------------------------------------------------------------
-#  release
-
-.PHONY: build-cross
-build-cross: LDFLAGS += -extldflags "-static"
-build-cross: $(GOX)
-	GO111MODULE=on CGO_ENABLED=0 $(GOX) -parallel=3 -output="_dist/{{.OS}}-{{.Arch}}/$(BINNAME)" -osarch='$(TARGETS)' $(GOFLAGS) -tags '$(TAGS)' -ldflags '$(LDFLAGS)' ./cmd/helm
-
-.PHONY: dist
-dist:
-	( \
-		cd _dist && \
-		$(DIST_DIRS) cp ../LICENSE {} \; && \
-		$(DIST_DIRS) cp ../README.md {} \; && \
-		$(DIST_DIRS) tar -zcf helm-${VERSION}-{}.tar.gz {} \; && \
-		$(DIST_DIRS) zip -r helm-${VERSION}-{}.zip {} \; \
-	)
-
-.PHONY: fetch-dist
-fetch-dist:
-	mkdir -p _dist
-	cd _dist && \
-	for obj in ${TARGET_OBJS} ; do \
-		curl -sSL -o helm-${VERSION}-$${obj} https://get.helm.sh/helm-${VERSION}-$${obj} ; \
-	done
-
-.PHONY: sign
-sign:
-	for f in _dist/*.{gz,zip,sha256} ; do \
-		gpg --armor --detach-sign $${f} ; \
-	done
-
-.PHONY: checksum
-checksum:
-	for f in _dist/*.{gz,zip} ; do \
-		shasum -a 256 "$${f}"  | awk '{print $$1}' > "$${f}.sha256" ; \
-	done
-
-# ------------------------------------------------------------------------------
-
-.PHONY: clean
+all: clean check-quality build test golangci
+
+ALL_PACKAGES=$(shell go list ./...)
+SOURCE_DIRS=$(shell go list ./... | cut -d "/" -f4 | uniq)
+
 clean:
-	@rm -rf $(BINDIR) ./_dist
-
-.PHONY: info
-info:
-	 @echo "Version:           ${VERSION}"
-	 @echo "Git Tag:           ${GIT_TAG}"
-	 @echo "Git Commit:        ${GIT_COMMIT}"
-	 @echo "Git Tree State:    ${GIT_DIRTY}"
+	rm -rf ./bin
+	GO111MODULE=on go mod tidy -v
+
+check-quality: setup lint fmt imports vet
+
+setup:
+	GO111MODULE=off go get -v golang.org/x/tools/cmd/goimports
+	GO111MODULE=off go get -v golang.org/x/lint/golint
+
+lint:
+	@if [[ `golint $(ALL_PACKAGES) | { grep -vwE "exported (var|function|method|type|const) \S+ should have comment" || true; } | wc -l | tr -d ' '` -ne 0 ]]; then \
+          golint $(ALL_PACKAGES) | { grep -vwE "exported (var|function|method|type|const) \S+ should have comment" || true; }; \
+          exit 2; \
+    fi;
+
+fmt:
+	gofmt -l -s -w $(SOURCE_DIRS)
+
+imports:
+	./scripts/lint.sh check_imports
+
+vet:
+	go vet ./...
+
+cyclo:
+	gocyclo -over 6 $(SOURCE_DIRS)
+
+
+fix_imports:
+	goimports -l -w .
+
+golangci:
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b bin/ v1.30.0
+	bin/golangci-lint run -v --deadline 5m0s
+
+
+build:
+	@echo "Building './bin/albatross'"
+	@mkdir -p ./bin
+	@go build -o bin/albatross ./cmd/albatross
```

Comment:
call the package as `cmd/server`

Reply:
The one concern I can think of with renaming the package as server is that go install with default options will result in a binary named service. We will be using Makefile/Dockerfile to build the binary for workflows, so that shouldn't matter much, but it might be slighly unintuitive I feel. Looked at other go repos as well, and they seem to be following a similar convention.